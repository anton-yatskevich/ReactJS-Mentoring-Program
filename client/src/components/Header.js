import React from 'react';
import PropTypes from 'prop-types';

const Header = ({ isSearchPage, goToSearchHandler }) => (
    <header>
        <h1>netflixroulette</h1>
        {!isSearchPage && <button type="button" onClick={goToSearchHandler}>SEARCH</button>}
    </header>
);

Header.propTypes = {
    isSearchPage: PropTypes.bool,
    goToSearchHandler: PropTypes.func
};

Header.defaultProps = {
    isSearchPage: null,
    goToSearchHandler: null
};

export default Header;
