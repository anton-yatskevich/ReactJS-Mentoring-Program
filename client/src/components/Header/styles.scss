.header {
	height: 60px;
	width: 100%;
	color: rgb(160, 0, 0);
	background-color: rgb(146, 133, 133);
	padding: 0 20px;
	border-bottom: 1px solid rgb(160, 0, 0);

	&__wrapper {
		max-width: 1200px;
		height: 100%;
		margin: 0 auto;
		display: flex;
		justify-content: space-between;
		align-items: center;
	}
}
