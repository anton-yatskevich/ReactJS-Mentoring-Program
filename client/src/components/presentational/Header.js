import { PureComponent } from 'react';

class Header extends PureComponent {
    render() {
        const { title } = this.props;
        return <header><h1>{title}</h1></header>;
    }
}

export default Header;
