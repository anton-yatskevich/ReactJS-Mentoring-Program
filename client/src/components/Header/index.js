import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash.noop';

import Button from '../Button';
import './styles.scss';

const Header = ({ isSearchPage, goToSearchHandler }) => (
    <header className="header">
        <div className="header__wrapper">
            <h1>netflixroulette</h1>
            {
                !isSearchPage && <Button onClickHandler={() => goToSearchHandler(null)} textValue="search" className="header__button" />
            }
        </div>
    </header>
);

Header.propTypes = {
    isSearchPage: PropTypes.bool,
    goToSearchHandler: PropTypes.func
};

Header.defaultProps = {
    isSearchPage: false,
    goToSearchHandler: noop
};

export default Header;
