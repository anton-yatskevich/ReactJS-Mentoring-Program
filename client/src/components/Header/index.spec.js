import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './index';

describe('Header component', () => {
    it('should be rendered with default props', () => {
        const HeaderComponent = shallow(<Header />);
        expect(toJson(HeaderComponent)).toMatchSnapshot();
    });
});
