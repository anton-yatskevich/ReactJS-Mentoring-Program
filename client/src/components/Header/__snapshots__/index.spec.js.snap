// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be rendered with default props 1`] = `
<header
  className="header"
>
  <div
    className="header__wrapper"
  >
    <h1>
      netflixroulette
    </h1>
    <Button
      className="header__button"
      isActive={false}
      onClickHandler={[Function]}
      textValue="search"
      value=""
    />
  </div>
</header>
`;
